src/utils/searchViewHelper.js

```diff
@@ -32,24 +47,35 @@ const searchViewHelper = {
     // If repository is selected after search value is input, fetch search results
     if (search !== '') {
       searchViewHelper.searchByRevisionOrEmail(repository, search);
+    } else {
+      // Fetch 10 most recent revisions when repository changes
+      store.dispatch(fetchRecentRevisions(repository));
     }
-    // Fetch 10 most recent revisions when repository changes
-    store.dispatch(fetchRecentRevisions(repository));
   },
 
   handleChangeSearch(event) {
     const search = event.target.value;
     const { repository } = store.getState().search;
     store.dispatch(updateSearchValue(search));
+    store.dispatch(updateSearchResults([]));
+    store.dispatch(updateInputError(false));
+    store.dispatch(updateInputHelperText(''));
+
+    const _idleTime = 500;
+    const _onTimeout = () => {
+      if (repository === '') {
+        store.dispatch(updateDropdownError(true));
+        store.dispatch(updateDropdownHelperText('Please select a repository'));
+      } else {
+        searchViewHelper.searchByRevisionOrEmail(repository, search);
+      }
+    };
+
+    if (_timeout) clearTimeout(_timeout);
 
     // If search input is cleared, clear results
     if (search === '') store.dispatch(updateSearchResults([]));
-
-    if (repository === '') {
-      console.log('Please select a repository');
-    } else {
-      searchViewHelper.searchByRevisionOrEmail(repository, search);
-    }
+    else _timeout = setTimeout(_onTimeout, _idleTime);
```

Comment:
nit: please use brackets for the else block.

I'm surprised that some eslint rule doesn't do that automatically :-)